pastis/matrix_generation/matrix_building_numerical.py

```diff
@@ -253,9 +263,30 @@ def calculate_unaberrated_contrast_and_normalization(instrument, design=None, re
         # Return the coronagraphic simulator (a tuple in the JWST case!)
         coro_simulator = jwst_sim
 
-    # Calculate coronagraph floor in dark hole
-    contrast_floor = util.dh_mean(coro_psf, dh_mask)
-    log.info(f'contrast floor: {contrast_floor}')
+        contrast_floor = util.dh_mean(coro_psf, dh_mask)
+        log.info(f'contrast floor: {contrast_floor}')
```

Comment:
The idea behind this block being outside of, and after all `if` statements for the instruments is that each if block should define a coronagraphic PSF and DH mask in the same format, so that we avoid duplicating code. The minimum I would ask you to do is to is to bring back this unindented block and delete duplications in the if blocks of LUVOIR, HiCAT and JWST.

Ideally, you would also write the if block for RST such that it can also calcualte the coro_floor in the same way, from a previously defined coro_psf and dh_mask.

Reply:
Ok, I have made up my mind about this now, see my main review comment.
In a nutshell for here, please do not use `raw_contrast()`, but instead use more modular functions that allow you to get the `norm` factor, the coronagraphic PSF, and the DH mask individually. Then you can also reinstate the original order of this block of code to calculate the contrast in this function, with the same block being used for all instruments.